pkg/bootstrapper/bootstrapper.go

```diff
@@ -499,63 +506,74 @@ func (wmcb *winNodeBootstrapper) createKubeletService() error {
 		Password:         "",
 		Description:      "OpenShift Kubelet",
 	}
+	// Get kubelet args
+	kubeletArgs := wmcb.createInitialKubeletArgs()
+
+	if wmcb.kubeletSVC == nil {
+		if err := wmcb.createKubeletService(c, kubeletArgs); err != nil {
+			return fmt.Errorf("failed to create kubelet service : %v ", err)
+		}
+	} else {
+		if err := wmcb.updateKubeletService(c, kubeletArgs); err != nil {
+			return fmt.Errorf("failed to update kubelet service : %v ", err)
+		}
+	}
+
+	if err := wmcb.kubeletSVC.setRecoveryActions(); err != nil {
+		return fmt.Errorf("failed to set recovery actions for Windows service %s : %v", KubeletServiceName, err)
+	}
+	return nil
+}
+
+// createKubeletService creates a new kubelet service to our specifications
+func (wmcb *winNodeBootstrapper) createKubeletService(c mgr.Config, kubeletArgs []string) error {
 	ksvc, err := wmcb.svcMgr.CreateService(KubeletServiceName, filepath.Join(wmcb.installDir, "kubelet.exe"), c, kubeletArgs...)
 	if err != nil {
 		return err
 	}
 
 	var dependents []*mgr.Service
-	if dependentSvc, err := wmcb.svcMgr.OpenService(kubeletDependentSvc); err == nil {
-		dependents = append(dependents, dependentSvc)
+	dependentSvc, err := wmcb.svcMgr.OpenService(kubeletDependentSvc)
+	if err != nil {
+		return fmt.Errorf("colud not get kubelet dependent services %s : %v", kubeletDependentSvc, err)
 	}
+	dependents = append(dependents, dependentSvc)
 	wmcb.kubeletSVC, err = newKubeletService(ksvc, dependents)
 	if err != nil {
 		return fmt.Errorf("could not initialize struct kubeletService: %v", err)
 	}
+	return nil
+}
 
-	if err := wmcb.kubeletSVC.setRecoveryActions(); err != nil {
-		return fmt.Errorf("failed to set recovery actions for Windows service %s", KubeletServiceName)
+// createKubeletService creates a new kubelet service to our specifications
+func (wmcb *winNodeBootstrapper) updateKubeletService(c mgr.Config, kubeletArgs []string) error {
+	// Stop the kubelet service as there could be open file handles from kubelet.exe on the plugin files
+	if err := wmcb.kubeletSVC.stop(); err != nil {
+		return fmt.Errorf("unable to stop kubelet service: %v", err)
 	}
 
-	return nil
-}
+	if err := configureKubelet(&c.BinaryPathName, kubeletArgs); err != nil {
+		return fmt.Errorf("error configuring kubelet service: %v", err)
+	}
 
-// refreshServiceManager will disconnect and reconnect from the Windows service API. In order to complete certain
-// operations, there must be zero handlers to the API present on the system.
-func (wmcb *winNodeBootstrapper) refreshServiceManager() error {
-	var err error
-	if err = wmcb.Disconnect(); err != nil {
```

Comment:
Do we need to call this after we finish configuration?

Reply:
We are also calling [refresh](https://github.com/openshift/windows-machine-config-bootstrapper/blob/master/pkg/bootstrapper/kubelet.go#L116) after configuration. Do we still go ahead and call this function ?

Reply:
I am going to defer to @sebsoto on this. I want to ensure we are closing all open handles.

Reply:
The refresh was important since we needed to close handles to the service manager in order to allow the service to be deleted. Now all we need to make sure to do is that we are closing it at the end. The process exiting should do that automatically, but to be safe we should explicitly disconnect

Reply:
Never assume exiting will clean things up automatically, especially on Windows ;-)

Reply:
Won`t the refresh function be required here since it calls, [svcobj.UpdateConfig](https://github.com/openshift/windows-machine-config-bootstrapper/blob/master/pkg/bootstrapper/kubelet.go#L121). We are also calling [Disconnect](https://github.com/openshift/windows-machine-config-bootstrapper/blob/master/cmd/bootstrapper/initialize_kubelet.go#L72) after IntializeKubelet, that should clean things up

Reply:
Yes. [ConfigureCNICmd](https://github.com/openshift/windows-machine-config-bootstrapper/blob/master/cmd/bootstrapper/configure_cni.go#L72)